onnx/defs/schema.h

```diff
@@ -735,6 +735,41 @@ class OpSchema final {
     return all_tensor_sequence_types;
   }
 
+  static const std::vector<std::string>& all_optional_types() {
```

Comment:
Is this used anywhere? I think it would be better NOT to introduce this kind of "limited" set of special types or encourage its use in op schemas.

Reply:
It is not used yet, but I guess it would be needed once introducing the ops and specifying input/output types.
What is your suggestion for this use-case?

Reply:
Once we introduce ops do we need to support all these types? can we remove this and simply add support for the set of types which are required for the op? Example optional(seq(tensor(complex64))) it seems unlikely that we will need support for this...